### PR TITLE
WIP: Add webhook user guides

### PR DESCRIPTION
**Pull request by lavalamp**
This PR starts user guides for webhook authors and webhook operators. I still need to add some cross referencing links.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6836)

### CODE REVIEW

File: docs/concepts/cluster-administration/admission-control-webhooks.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+### What does your webhook trust?
+
+If the webhook is going to run in your cluster, the documentation for the
+webhook should tell you how to set up the identities it will trust. If the
+webhook is an external service, the service provider will have to give you some
+mechanism for demonstrating an identity (e.g., a token or a client-cert for
+mutual TLS).
+
+The configuration file for admission controllers allows you to specify a
```
Comment (caesarxuchao) [inline docs/concepts/cluster-administration/admission-control-webhooks.md]: It's worth mentioning that the configuration file is specified via the `--admission-control-config-file` flag of the apiserver. I hope that we had a doc on this config file.

File: docs/concepts/cluster-administration/admission-control-webhooks.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+file in "kubeconfig" format. Each entry has a name; if the name "matches" (see
+below) a webhook, the contents of that entry will be used as the identity when
+calling out to that webhook. What does "match" mean? It is DNS based; e.g.
+`*.bar.example.com` matches `foo.bar.example.com`, but if a
+`foo.bar.example.com` section is present, it would take precendence. If the
+webhook is running in the cluster, the name and namespace of the service will be
+put into the form "name.namespace.svc" before matching.
+
+If your cluster is self-hosted, it likely makes sense to put these files into a
```
Comment (caesarxuchao) [inline docs/concepts/cluster-administration/admission-control-webhooks.md]: What does "these files" refer to? The tokens and client-certs?

File: docs/concepts/cluster-administration/admission-control-webhooks.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+order to tell if they need to call a webhook. You'll need to make sure they have
+appropriate permissions.
+
+## Bootstrapping
+
+It is important to understand what a webhook will do and where it is
+implemented. For example, suppose a webhook is running via a deployment and
+modifies all pod objects. Then it is very important to be able to create the
+webhook's own pods when the webhook is not running, otherwise the cluster will
+be wedged. Do this by using the namespace label selector.
```
Comment (caesarxuchao) [inline docs/concepts/cluster-administration/admission-control-webhooks.md]: How about adding a [link](https://github.com/kubernetes/api/blob/release-1.9/admissionregistration/v1beta1/types.go#L191) to the namespaceSelector field? A reader might not know what it it.

**Comment by heckj** [comment]
@lavalamp wanted to point out the CI failures - adding the documents requires adding some entries to the Table of contents YAML files as well. You can see the details at https://travis-ci.org/kubernetes/website/builds/324832783

**Comment by erictune** [comment]
lgtm for eng

**Comment by zacharysarah** [comment]
👋 We've migrated the site's framework to Hugo. Please re-open this PR with changes to files in their post-migration location (`/content/en/docs/...`).

**Pull request closed without merging**